tests/test_em_fdem_electric_dipole.py

```diff
@@ -0,0 +1,334 @@
+        )
+
+        e = edws.electric_field(xyz)
+        print(
+            "\n\nTesting Electric Dipole {} orientation\n".format("45 degree")
+        )
+
+        self.compare_fields(e, np.vstack([extest, eytest, eztest]).T)
+
+
+class TestFDEMdipole_SimPEG(unittest.TestCase):
+
+    tol = 1e-1 # error must be an order of magnitude smaller than results
+
+    # put the source at the center
+
+    def getFaceSrc(self, mesh):
+        csx = mesh.hx.min()
+        csz = mesh.hz.min()
+        srcInd = (
+            (mesh.gridFz[:, 0] < csx) &
+            (mesh.gridFz[:, 2] < csz/2.) & (mesh.gridFz[:, 2] > -csz/2.)
+        )
+
+        src_vecz = np.zeros(mesh.nFz, dtype=complex)
+        src_vecz[srcInd] = 1.
+
+        return np.hstack(
+            [np.zeros(mesh.vnF[:2].sum(), dtype=complex), src_vecz]
+        )
+
+    def getProjections(self, mesh):
+        ignore_inside_radius = 5*mesh.hx.min()
+        ignore_outside_radius = 40*mesh.hx.min()
+
+        def ignoredGridLocs(grid):
+            return (
+                (
+                    grid[:, 0]**2 + grid[:, 1]**2 + grid[:, 2]**2  <
+                    ignore_inside_radius**2
+                ) | (
+                    grid[:, 0]**2 + grid[:, 1]**2 + grid[:, 2]**2 >
+                    ignore_outside_radius**2
+                )
+            )
+
+        # Faces
+        ignore_me_Fx = ignoredGridLocs(mesh.gridFx)
+        ignore_me_Fz = ignoredGridLocs(mesh.gridFz)
+        ignore_me = np.hstack([ignore_me_Fx, ignore_me_Fz])
+        keep_me = np.array(~ignore_me, dtype=float)
+        Pf = discretize.utils.sdiag(keep_me)
+
+        # Edges
+        ignore_me_Ey = ignoredGridLocs(mesh.gridEy)
+        keep_me_Ey = np.array(~ignore_me_Ey, dtype=float)
+        Pe = discretize.utils.sdiag(keep_me_Ey)
+
+        return Pf, Pe
+
+    def test_e_dipole_v_SimPEG(self):
+
+        def compare_w_SimPEG(name, geoana, simpeg):
+
+            norm_geoana = np.linalg.norm(geoana)
+            norm_simpeg = np.linalg.norm(simpeg)
+            diff = np.linalg.norm(geoana - simpeg)
+            passed = diff < self.tol * 0.5 * (norm_geoana + norm_simpeg)
+            print(
+                "  {} ... geoana: {:1.4e}, SimPEG: {:1.4e}, diff: {:1.4e}, "
+                "passed?: {}".format(
+                    name, norm_geoana, norm_simpeg, diff, passed
+                )
+            )
+
+            return passed
+
+        print("\n\nComparing Electric dipole with SimPEG")
+
+        sigma_back = 1e-1
+        freqs = np.r_[10., 100.]
+
+        csx, ncx, npadx = 0.5, 40, 20
+        ncy = 1
+        csz, ncz, npadz = 0.5, 40, 20
+
+        hx = discretize.utils.meshTensor(
+            [(csx, ncx), (csx, npadx, 1.2)]
+        )
+        hy = 2*np.pi / ncy * np.ones(ncy)
+        hz = discretize.utils.meshTensor(
+            [(csz, npadz, -1.2), (csz, ncz), (csz, npadz, 1.2)]
+        )
+
+        mesh = discretize.CylMesh([hx, hy, hz], x0='00C')
```

Comment:
I like how we compare to numerical solutions in this repo, and backwards in simpeg. Ha.